[documentation] Start an event preparation guide (#3522)
2i2c-org/infrastructure: docs/howto/event-prep.md

A hub's specific setup is usually optimized based on the day to day usage expectations. But because events usually imply a different usage pattern, the infrastructure might need to be adjusted in order to accommodate the spikes in activity.

```{important}
The communities we serve have the responsibility to notify us about an event they have planned on a 2i2c hub [at least three weeks before](https://docs.2i2c.org/community/events/#notify-the-2i2c-team-about-the-event) the event will start. This should allow us enough time to plan and prepare the infrastructure for the event properly if needed.
```

The events might vary in type, so the following list is not complete and does not cover all of them (yet). Most common event types are exams, workshops etc.

## Event checklist

Below are listed the main aspects to consider adjusting on a hub to prepare it for an event:

### 1. Quotas

We must ensure that the quotas from the cloud provider are high-enough to handle expected usage. It might be that the number of users attending the event is very big, or their expected resource usage is big, or both. Either way, we need to check the the existing quotas will accommodate the new numbers.

> ```{admonition} Action to take
> :class: tip

Reviewer: ````suggestion
```{tip} Action to take
````

Does this work?

Author: Unfortunately "Tip" is still the name of the admonition. This is how it looks like in both versions:

<img width="814" alt="Screenshot 2023-12-13 at 15 39 20" src="https://github.com/2i2c-org/infrastructure/assets/7579677/c6b8ad45-ee17-4233-9d42-34c9ff0a4132">
<img width="814" alt="Screenshot 2023-12-13 at 15 39 34" src="https://github.com/2i2c-org/infrastructure/assets/7579677/fec1bb38-02f0-413a-9742-c3d9c2d5a592">

Reviewer: Ohhhh, sorry, I get what you're trying to do now

- follow the [AWS quota guide](hub-deployment-guide:cloud-accounts:aws-quotas) for information about how to check the quotas in an AWS project

> - follow [GCP quota guide](hub-deployment-guide:cloud-accounts:aws-quotas) for information about how to check the quotas in a GCP project

Reviewer: ```suggestion
- follow the [GCP quota guide](hub-deployment-guide:cloud-accounts:aws-quotas) for information about how to check the quotas in a GCP project
```

for consistency with the bullet above

```

### 2. Consider dedicated nodepools on shared clusters

> If the hub that's having an event is running on a shared cluster, then we might want to consider putting it on a dedicated nodepool as that will help with cost isolation, scaling up/down effectively, not impacting other hub's users performance.

Reviewer: ```suggestion
If the hub that's having an event is running on a shared cluster, then we might want to consider putting it on a dedicated nodepool as that will help with cost isolation, scaling up/down effectively, and avoid impacting other hub's users performance.
```

    ```
    ````

4. **Don't oversubscribe resources**
    The oversubscription factor is how much larger a limit is than the actual request (aka, the minimum guaranteed amount of a resource that is reserved for a container). When this factor is greater, then a more efficient node packing can be achieved because usually most users don't use resources up to their limit, and more users can fit on a node.

    However, a bigger oversubscription factor also means that the users that use more resources than they are guaranteed can get their kernels killed or CPU throttled at some other times, based on what other users are doing. This inconsistent behavior is confusing to end users and the hub, so we should try and avoid this during events.

    ````{admonition} Action to take
    :class: tip

    For an event, you should consider an oversubscription factor of 1. For this you can use the deployer script by passing it the instance type where the pods will be scheduled on (in this example is `n2-highmem-4`), then from its output, pick the choice(s) that will be used during the event based on expected usage.

    ```bash
    deployer generate resource-allocation choices n2-highmem-4

>     ```

Reviewer: It might be nice to expand this example a bit by also showing the output of this command and flagging which option would be best for this scenario?

Author: @sgibson91, I did this in https://github.com/2i2c-org/infrastructure/pull/3522/commits/6f94462d731ac00e315271d47fa6c1d6789b2eff

    ````

#### 3.2. Setting a minimum node count on a specific node pool

> TODO

Reviewer: ````suggestion
```{warning}
This section is a Work in Progress!
```
````

I'd maybe make this a callout so when it's rendered it's a bit clearer? Same below.
